stock_cycle_count/models/stock_location.py

```diff
@@ -21,13 +21,12 @@
 class StockLocation(models.Model):
     _inherit = "stock.location"
 
-    @api.multi
     def _compute_loc_accuracy(self):
         for rec in self:
-            history = self.env["stock.inventory"].search(
-                [("location_id", "=", rec.id), ("state", "=", "done")]
-            )
-            history = history.sorted(key=lambda r: r.write_date, reverse=True)
+            history = self.env["stock.inventory"].search([("state", "=", "done")])
+            history = history.filtered(
+                lambda x: rec.id in x.mapped("location_ids.id")
+            ).sorted(key=lambda r: r.write_date, reverse=True)
```

Comment:
I think this can be done in one search, something like:

```python
history = self.env["stock.inventory"].search([
    ("location_ids", "in", rec.id), ("state", "=", "done")], order="write_date desc")
```

Can you test this?